### DISCUSSION TITLE
use nerdctl in the container to access the containerd.sock of the host

### DISCUSSION THREAD

**Discussion by qi0523** [containerd/nerdctl#1383] [Q&A] [answered]
Hardware and software environment as follows:

`Ubuntu-20.04.3`
`containerdv1.5.11`
`nerdctlv0.23.0`

I use `nerdctl run -it -v /run/containerd/containerd.sock:/run/containerd/containerd.sock -v /usr/local/bin/nerdctl:/usr/local/bin/nerdctl --name=myubuntu1 ubuntu:latest` to run a container.

Then, i use `nerdctl --debug run -d --name=myubuntu2 ubuntu:latest` to run a new container and debug in myubuntu1 container.  The error messages as follows: 

`root@a59268d28a64:~# nerdctl --debug run -d --name=myubuntu2 ubuntu:latest`

`DEBU[0000] failed to run [aa-exec -p nerdctl-default -- true]: ""  error="exec: \"aa-exec\": executable file not found in $PATH"`

`DEBU[0000] verification process skipped `         
   
`DEBU[0000] Failed to unmount snapshot "/tmp/initialC3307108135" `

`FATA[0000] operation not permitted`

How should i do to use nerdctl in the container to access the containerd.sock of the host and create container?

**Comment by qi0523** [comment]
Lately, i use `--privileged`, give extended privileges to myubuntu1 container. 

`nerdctl run -it --privileged --name=myubuntu1 -v /run/containerd/containerd.sock:/run/containerd/containerd.sock -v /usr/local/bin/nerdctl:/usr/local/bin/nerdctl -v /usr/bin/aa-exec:/usr/bin/aa-exec ubuntu:latest`

Then, i use `nerdctl --debug run -d --name=myubuntu2 ubuntu:latest` to debug and create a new container. The error messages are as follow:
`root@3ce1eb02ce56:~# nerdctl --debug run -d --name=myubuntu2 ubuntu:latest`
`DEBU[0000] verification process skipped`       
`FATA[0000] no such file or directory`

I use `journalctl -u containerd` to check the containerd's logs. However, the `nerdctl --debug run -d --name=myubuntu2 ubuntu:latest` doesn't generate logs.  I am not sure if nerdctl had connected to containerd.sock.

**Comment by fahedouch** [comment, marked as answer]
plz take a look on this https://github.com/containerd/nerdctl/discussions/1090#discussioncomment-2897738

**Reply by qi0523** [reply]
Thanks.

`myubuntu2` container has successfully run inside `myubuntu1` container.

However, i use `nerdctl inspect myubuntu2` to check the container's `NetworkSettings` inside `myubuntu1` container . The `NetworkSettings` info is `null`.

I repeat  `nerdctl inspect myubuntu2`  multiple times. It may have been a success. The error messages are as follows:

`WARN[0000] failed to inspect NetNS                       error="Failed to open current namespace: failed to Statfs \"/proc/28/task/28/ns/net\": no such file or directory" id=ad75c58916af00c0488da780239535aaae33ec92e3fbdef8de1190301c2ff36e`

![21](https://user-images.githubusercontent.com/42767692/191416506-f243c0fa-5ebc-4771-a497-df2f2608ad69.png)

**Reply by fahedouch** [reply]
can you open an issue ticket for this random behaviour plz
